Code/tests/test_filefinder.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
+import plistlib
+import StringIO
+import unittest
+
+import mock
+from autopkglib import ProcessorError
+from autopkglib.FileFinder import FileFinder
+
+
+class TestFileFinder(unittest.TestCase):
+    """Test class for FileFinder Processor."""
+
+    def setUp(self):
+        self.good_env = {"find_method": "glob", "pattern": "test"}
+        self.bad_env = {"find_method": "fake"}
+        self.input_plist = StringIO.StringIO(plistlib.writePlistToString(self.good_env))
+        self.processor = FileFinder(infile=self.input_plist)
+
+    def tearDown(self):
+        self.input_plist.close()
+
+    def test_raise_if_not_glob(self):
+        """Raise an exception if glob is not passed to find_method."""
+        self.processor.env = self.bad_env
+        with self.assertRaises(ProcessorError):
+            self.processor.main()
+
+    @mock.patch("autopkglib.FileFinder.globfind")
+    def test_no_fail_if_good_env(self, mock_glob):
+        """The processor should not raise any exceptions if run normally."""
+        self.processor.env = self.good_env
+        mock_glob.return_value = "test"
+        self.processor.main()
+
+    @mock.patch("autopkglib.FileFinder.globfind")
+    def test_found_a_match(self, mock_glob):
+        """If we find a match, it should be in the env."""
+        self.processor.env = self.good_env
+        mock_glob.return_value = "test"
+        self.processor.main()
+        self.assertEqual(self.processor.env["found_filename"], "test")
+
+    @mock.patch("autopkglib.FileFinder.unmount")
```

Comment:
order to test.This is an example of patching things that actually change system state - like mounting and unmounting a DMG. We patch the mount and unmount functions here, and predefine their return values to provide a fixed path. That way, they don't depend on actual DMGs being present in order to test.